modules/programs/helix.nix

```diff
@@ -136,10 +173,29 @@ in {
         "helix/config.toml" = mkIf (cfg.settings != { }) {
           source = tomlFormat.generate "helix-config" cfg.settings;
         };
-        "helix/languages.toml" = mkIf (cfg.languages != [ ]) {
-          source =
-            tomlFormat.generate "helix-config" { language = cfg.languages; };
-        };
+        "helix/languages.toml" = mkIf (cfg.languages != [ ] || cfg.grammars
+          != [ ] || cfg.use-grammars != { }) {
+            source =
+              # NB: helix requires that `use-grammars` be the first key in languages.toml if present
+              # pkgs.formats.toml relies on piping generated json through
+              # [remarshal](https://github.com/remarshal-project/remarshal),
+              # so we have to concatenate in order to preserve this ordering
```

Comment:
Are you sure about that? AFAIK helix uses the `serde_derive` implementation for it which should respect the TOML standard, no weird TOML breaking custom deserializer (in the way that it relies on ordered tables/objects).

Reply:
The docs say
> `# Note: this key must come **before** the [[language]] and [[grammar]] sections`
but indeed this just means that it should be *outside* of the `language` and `grammar` sections, so just regular TOML.